filters_test.go

```diff
@@ -346,6 +348,68 @@ func TestMatch(t *testing.T) {
 	}
 }
 
+func TestReadInput(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		Description string
+		Input       string
+		Default     string
+		Want        string
+		ShouldFail  bool
+	}{
+		{
+			Description: "User Input OK",
+			Input:       "123",
+			Default:     "Unused",
+			Want:        "123",
+		},
+		{
+			Description: "No User Input",
+			Input:       "",
+			Default:     "default",
+			Want:        "default",
+		},
+		{
+			Description: "Complex Input",
+			Input:       "/user/test @@@ testing one two three",
+			Default:     "default",
+			Want:        "/user/test @@@ testing one two three",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.Description, func(t *testing.T) {
+			// Prepare Stdin before calling prompt.
+			tmp, err := ioutil.TempFile("", tc.Description)
```

Comment:
A `bytes.Buffer` would probably be more convenient here, wouldn't it?

Reply:
For which function exactly? Both functions `t.Run` and `ioutil.Tempfile` take `string`.

```go
func (t *T) Run(name string, f func(t *T)) bool 
func TempFile(dir, pattern string) (f *os.File, err error)
```

Please, correct me if I misunderstood.

Reply:
Well, the point is that we just need some `io.Reader`, isn't it? We could use a disk file for this purpose, but it's tremendously slow compared to anything in memory (like a `bytes.Buffer`), and it also requires us to have a writeable filesystem, which isn't necessarily always the case (think about running tests in a container, for example).